plugin/pkg/scheduler/generic_scheduler.go

```diff
@@ -94,7 +113,12 @@ func (g *genericScheduler) Schedule(pod *api.Pod, nodeLister algorithm.NodeListe
 		}
 	}
 
-	return g.selectHost(priorityList)
+	selectNode, err := g.selectHost(priorityList)
+	// If binding pod to a node, invalid cache of this node.
+	if err == nil {
+		g.cache.InvalidNodeCache(selectNode)
```

Comment:
It seems you just invalid node cache here, but I think it's not enough. At least in the following cases, we should invalidate cache too:

1) node updated(eg: labels updated, node has been deleted, node is lost, etc)
2) a bound pod updated(eg: a bound pod has been deleted, etc)

We can invalidate cache at watch time(Addfunc/DeleteFuncs/UpdateFuncs)